### PR TITLE
Retire torch.distributed.pipeline

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #127354

Actually retiring module after deprecation warning for a while.
The new supported module is: torch.distributed.pipelining. 
Please migrate.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @ezyang @gchanan @albanD

### CODE REVIEW

**Review by wconstab — APPROVED** [review]
oops i submitted a parallel PR to remove old pipeline.rst. ill cancel mine.

File: docs/source/conf.py [hunk 1]
```diff
@@ -606,29 +606,22 @@
     # torch.distributed.optim.utils
     "as_functional_optim",
     "register_functional_optim",
-    # torch.distributed.pipeline.sync.checkpoint
```
Comment (albanD) [inline docs/source/conf.py:609]: Please delete the corresponding entries! Not just the comment saying which module they come from!

**Review by albanD — CHANGES_REQUESTED** [review]

File: docs/source/conf.py (hunk 1 above)
Comment (kwen2501) [reply docs/source/conf.py:609]: hahahaha, sorry my bad, I thought those are real entries, but just commented out :)

**Pull request closed without merging**